### PR TITLE
[Merged by Bors] - TY-2849 update stack_id for items we pass to the exploration stack

### PR DESCRIPTION
**Pull request by Robert-Steiner**
Ticket: 
- [TY-2849]

[TY-2849]: https://xainag.atlassian.net/browse/TY-2849?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/core/src/stack/exploration.rs [hunk 1]
```diff
@@ -64,8 +64,16 @@ impl Stack {
...
-        Self::validate_documents_stack_id(new_documents, Stack::id())?;
-        let documents = chain!(&self.data.documents, new_documents)
+        let new_documents = new_documents
+            .iter()
+            .cloned()
+            .map(|mut doc| {
+                doc.stack_id = Self::id();
+                doc
+            })
+            .collect_vec();
```
Comment (acrrd) [inline discovery_engine_core/core/src/stack/exploration.rs:74]: :grey_question: Is `collect_vec` really needed? We should be able to chain this iterator with an iterator from `self.data.documents`, something like: `self.data.documents.iter().cloned().chain(new_documents).collect()`.

Comment (acrrd) [reply discovery_engine_core/core/src/stack/exploration.rs:74]: This will also avoid to clone two times the documents from `new_documents`.

Comment (Robert-Steiner) [reply discovery_engine_core/core/src/stack/exploration.rs:74]: Indeed your suggestion is better 👍
i adjusted it

**Review by acrrd — APPROVED** [review]

**Comment by acrrd** [comment]
bors merge

**Pull request closed without merging**